src/messages.lisp

```diff
@@ -276,6 +276,30 @@
       :required nil))
   :documentation "Native Quil and associated metadata returned from quilc.")
 
+(defmessage |RewriteArithmeticRequest|
+    ((|quil|
+      :documentation "Native Quil for which to rewrite arithmetic parameters."
+      :type :string
+      :required t))
+  :documentation "A request type to handle compiling arithmetic out of gate parameters.")
+
+(defmessage |RewriteArithmeticReponse|
+    ((|quil|
+      :documentation "Native Quil rewritten with no arithmetic in gate parameters."
+      :type :string
+      :required t)
+
+     (|original_memory_descriptors|
+      :documentation "The declared memory descriptors in the Quil of the related request."
+      :type (:list :any)
```

Comment:
Similarly, I'd prefer that this have a more descriptive type. I think `(:list |ParameterSpec|)` would be appropriate.

Reply:
Awesome! thank you. It might be better to have it match `memory_descriptors` and be a `(:map :string -> |ParameterSpec|)` but I'm not sure